pkg/plugins/gateway/algorithms/prefix_cache.go

```diff
@@ -18,113 +18,205 @@ package routingalgorithms
 
 import (
 	"fmt"
+	"math"
 	"math/rand"
+	"sort"
 	"strconv"
 
-	"github.com/vllm-project/aibrix/pkg/plugins/gateway/prefixcacheindexer"
-	"github.com/vllm-project/aibrix/pkg/plugins/gateway/prefixcacheindexer/tokenizer"
+	"github.com/vllm-project/aibrix/pkg/cache"
+	"github.com/vllm-project/aibrix/pkg/plugins/gateway/algorithms/prefixcacheindexer"
+	"github.com/vllm-project/aibrix/pkg/plugins/gateway/algorithms/tokenizer"
 	"github.com/vllm-project/aibrix/pkg/types"
 	"github.com/vllm-project/aibrix/pkg/utils"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/klog/v2"
 )
 
-var (
-	RouterPrefixCache types.RoutingAlgorithm = "prefix-cache"
-)
-
-func init() {
-	RegisterDelayedConstructor(RouterPrefixCache, NewPrefixCacheRouter)
-}
-
 const (
-	defaultPrefixCacheMatchThresholdPercent = 50
+	defaultPodRunningRequestImbalanceAbsCount = 8
```

Comment:
what's the magic number from?

Reply:
Its a magic number, no theory behind deriving it. While reading other open-source routers, I came across sglang implementation and it uses abs value of 32 for load balancing. 

It is configurable value and leave it to user to optimize it based off their workload and GPU type.

For 4K input token and 2k output token, 8 provided a appropriate balance for A10 and A100. Higher number than 8, I found results to degrade.